### PR TITLE
fix: json plugin should not log sensitive data by default

### PR DESCRIPTION
**Pull request by Shurtu-gal**
Fixes: amplication/amplication#8016

## PR Details

- Added flexibility for users to choose which fields they want to log, mask or ignore

## PR Checklist

- [ ] Tests for the changes have been added
- [x] `npm build` doesn't throw any error
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: plugins/logger-json/src/templates/logger.config.ts [hunk 1]
```diff
@@ -3,21 +3,98 @@ import { Params } from "nestjs-pino";
 
 ADDITIONAL_LOG_PROPERTIES_KEY;
 
+const defaultSerializer = (
```
Comment (overbit) [inline plugins/logger-json/src/templates/logger.config.ts:6]: although this implementation is really smart, I'd recommend to simplify it by leveraging the nestjs-pino library configuration that allow to achieve the same by configuration. Check here for more details https://getpino.io/#/docs/redaction

**Review by overbit — CHANGES_REQUESTED** [review]

File: plugins/logger-json/src/templates/logger.config.ts (hunk 1 above)
Comment (Shurtu-gal) [reply plugins/logger-json/src/templates/logger.config.ts]: Done @overbit.

File: plugins/logger-json/src/templates/logger.config.ts [hunk 2]
```diff
@@ -6,18 +6,32 @@ ADDITIONAL_LOG_PROPERTIES_KEY;
...
   return {
     pinoHttp: {
       level: logLevel,
+      transport: {
+        target: "pino-pretty",
+        options: {
+          colorize: true,
+          ignore: "pid,hostname",
+        },
+      },
```
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply plugins/logger-json/src/templates/logger.config.ts:29]: Yeah, I meant to add it as a opt in feature in dev environment. Will rectify it.

[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply plugins/logger-json/src/templates/logger.config.ts]: @overbit can you check if it's fine now? Or should I remove it completely?

**Review by overbit — APPROVED** [review]
Nice! 💯

**Pull request merged**